force-app/main/default/classes/TaskTriggerHandler.cls

```diff
@@ -1,53 +1,47 @@
 public without sharing class TaskTriggerHandler {
    
     public static void checkOnCompleteTask(List<Task> taskList){
-        List<Contact> toUpdate = [Select Id, Name, Points__c,
-                                  	(
-                                    	Select Id,ActivityDate,CreatedDate ,Task_Type__c,Points__c,IsClosed
-                                        FROM Tasks
-                                        WHERE Id IN :taskList
-                                        AND IsClosed = true
-                                    )	
-                                  FROM Contact];
+        List<Contact> selectedContacts = [Select Id, Name, Points__c,
+                                        (
+                                            Select Id,ActivityDate,CreatedDate ,Task_Type__c,Points__c,IsClosed
+                                            FROM Tasks
+                                            WHERE Id IN :taskList
+                                            AND IsClosed = true
+                                        )	
+                                      	FROM Contact];
         
-        for(Contact c : toUpdate)
+		List<Contact> toUpdate = new List<Contact>();
```

Comment:
you could improve the name of this variable (in future commits). something like: contactsToUpdate